### DISCUSSION TITLE
Question about message at the end of python installation

### DISCUSSION THREAD

**Discussion by quazi-irfan** [Homebrew/discussions#4041] [Everyday usage]
I am under the impression that if I do `brew install x` then homebrew installs the package at `/opt/homebrew/Celler/x`, and the binary is symlinked to `/opt/homebrew/bin`. I have installed python3.9 and now have python3.9 symlink available at `/opt/homebrew/bin`, which is pointing to `/opt/homebrew/Celler/python@3.9/3.9.16/bin/python3.9`. So far so good.

But at the end of python3.9 installation, I saw this message that I didn't understand,
```
Unversioned and major-versioned symlinks `python`, `python3`, `python-config`, `python3-config`, `pip`, `pip3`, etc. pointing to `python3.9`, `python3.9-config`, `pip3.9` etc., respectively, have been installed into 
  /opt/homebrew/opt/python@3.9/libexec/bin 
```
I have the following questions about this message,
1. Why is homebrew installing package content outside of `/opt/homebrew/Celler`?
2. What are these inversion and major-versioned symlinks being referred to? 
3. How does homebrew prevent collision with the default system python installation?

**Comment by SMillerDev** [comment]
> Why is homebrew installing package content outside of `/opt/homebrew/Cellar`

`/opt/homebrew/opt/python@3.9` is symlinked to `/opt/homebrew/Cellar/python@3.9/<latest_version_here>` so that you don't need to update any usage when software updates a version.

> What are these inversion and major-versioned symlinks being referred to?

Exactly what is in the list there. Symlinks for python that only indicate a major version or no version at all.

> How does homebrew prevent collision with the default system python installation?

By installing somewhere else.
